### PR TITLE
Multistage builds for netplan.io

### LINKED ISSUE #77: Multistage builds for netplan.io
## Summary

- [x] Multistage builds to decrease image size, faster image building.
- [x] Update `.dockerignore` to ignore files uncessary in prod
- [x] Update image to Ubuntu Focal

### PR DESCRIPTION
**Pull request by carkod**
## Done

Multistage builds to decrease image size, faster image building.

## QA

- `export DOCKER_BUILDKIT=1`
- Build image with new dockerfile `docker build --tag myimage --build-arg BUILD_ID=myfakeid .`
- Run conainter with this image `docker run -ti -p 8111:80 myimage`
- Check build_id `myfakeid` is present `curl localhost:8111`

## Issue / Card

Fixes #77

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -9,6 +9,15 @@
...
 HACKING.md
+renovate.json
+LICENCE
+.sass-lint.yml
+troubleshooting.md
+reference.md
+examples.md
+faq.md
+design.md
+sitemap.xml
```
Comment (jpmartinspt) [inline .dockerignore]: This is needed.

File: .dockerignore [hunk 2]
```diff
@@ -9,6 +9,15 @@
 cache.sqlite
 README.md
 HACKING.md
+renovate.json
+LICENCE
+.sass-lint.yml
```
Comment (jpmartinspt) [inline .dockerignore]: I think we need the sass-lint file otherwise sass linting won't work with `./run`? Please test

Comment (carkod) [reply .dockerignore]: Nope. We are not using sass-lint anywhere in the build (package.json) therefore, multistage builds isn't using it either.

In any case, I have QA'ed with this and builds successfully.

File: Dockerfile [hunk 3]
```diff
@@ -1,17 +1,56 @@
...
+
+# Set up environment
+ENV LANG C.UTF-8
+WORKDIR /srv
+
+# Install nginx
+RUN apt-get update && apt-get install --no-install-recommends --yes nginx
+
+# Import code, build assets and mirror list
+RUN rm -rf package.json yarn.lock .babelrc webpack.config.js Gemfile Gemfile.lock nginx.conf requirements.txt
```
Comment (tbille) [inline Dockerfile]: The `requirements.txt` should be in the `.dockerignore` file since it's only used in the circle CI job.

File: Dockerfile [hunk 4]
```diff
@@ -1,17 +1,56 @@
...
+
+ARG BUILD_ID
 ADD nginx.conf /etc/nginx/sites-enabled/default
-RUN sed -i "s/~REVISION_ID~/${REVISION_ID}/" /etc/nginx/sites-enabled/default
+RUN sed -i "s/~BUILD_ID~/${BUILD_ID}/" /etc/nginx/sites-enabled/default
 
 STOPSIGNAL SIGTERM
 
 CMD ["nginx", "-g", "daemon off;"]
+
```
Comment (tbille) [inline Dockerfile]: Hello extra line :)
You can remove it.

**Comment by tbille** [comment]
A small comment. Then I am happy to merge.

**Comment by carkod** [comment]
Let's get this approved please. Do not merge, or if you do merge, update jenkins job config.

**Pull request merged**